src/System.Console/src/System/Console.cs

```diff
@@ -157,29 +134,26 @@ public static bool IsInputRedirected
         {
             get
             {
-                StrongBox<bool> redirected = Volatile.Read(ref _isStdInRedirected) ??
-                    EnsureInitialized(ref _isStdInRedirected, () => new StrongBox<bool>(ConsolePal.IsInputRedirectedCore()));
-                return redirected.Value;
+                EnsureInitialized(ref _isStdInRedirected, () => new StrongBox<bool>(ConsolePal.IsInputRedirectedCore()));
+                return _isStdInRedirected.Value;
```

Comment:
Nit: it'd be better to store the result of EnsureInitialized into a local, as the previous code had it, e.g.
```C#
var redirected = EnsureInitialized(...);
return redirect.Value;
```